libmscore/tremolo.cpp

```diff
@@ -234,71 +235,38 @@ void Tremolo::layoutOneNoteTremolo(qreal x, qreal y, qreal _spatium)
-                        { 6 - 4 * .8,  3          },  // line 3
-                        { 2         ,  3          }   // line 4
-                        },
-                     // UP
-                     {
-                        // even line   odd line
-                        { -6,          -5          },  // line 1
-                        { -6,          -5          },  // line 2
-                        { -6,          -3 - 4 * .8 },  // line 3
-                        { -2 - 6 * .8, -3 - 6 * .8 }   // line 4
-                        }
-                     },
-                  // stem with hook
-                  {
-                     // DOWN
-                     {
-                        // even line   odd line
-                        { 3,           3          },  // line 1
-                        { 2,           2          },  // line 2
-                        { 2,           2          },  // line 3
-                        { 2,           2          }   // line 4
-                        },
-                     // UP
-                     {
-                        // even line   odd line
-                        { -3,          -3          },  // line 1
-                        { -2 - 2 * .8, -2 - 2 * .8 },  // line 2
-                        { -2 - 4 * .8, -2 - 4 * .8 },  // line 3
-                        { -2 - 6 * .8, -2 - 6 * .8 }   // line 4
-                        }
-                     },
-                  // stem with beam
-                  {
-                     // DOWN
-                     {
-                        // even line   odd line
-                        { 3,           3          },  // line 1
-                        { 2,           2          },  // line 2
-                        { 2,           2          },  // line 3
-                        { 2,           2          }   // line 4
-                        },
-                     // UP
-                     {
-                        // even line   odd line
-                        { -3,          -3          },  // line 1
-                        { -2 - 2 * .8, -2 - 2 * .8 },  // line 2
-                        { -2 - 4 * .8, -2 - 4 * .8 },  // line 3
-                        { -2 - 6 * .8, -2 - 6 * .8 }   // line 4
-                        }
-                     },
-                  };
-            int idx = _chord1->hook() ? 1 : (_chord1->beam() ? 2 : 0);
-            y = (line + t[idx][up][_lines-1][line & 1]) * .5 * _spatium;
+            const qreal td = score()->styleS(Sid::tremoloDistance).val();
+            const qreal sw = score()->styleS(Sid::tremoloStrokeWidth).val();
+
+            qreal t;
+            // nearest distance between note and tremolo stroke should be no less than 3.0
+            if (chord()->hook() || chord()->beam()) {
+                  t = up ? -3.0 - (2.0 * (lines() - 1)) * td - 2.0 * sw : 3.0;
+                  }
+            else { 
+                  if      (!up && !(line & 1)) // stem is down; even line
```

Comment:
indent

Reply:
I intend to align the conditionals, in this way it's clearer.